### PR TITLE
Upgrade rails 7.1.2

### PR DESCRIPTION
**Pull request by jyoung-gds**
[Trello ticket link](https://trello.com/c/3S3PjGTD/1810-upgrade-contacts-admin-to-rails-712)

### Changes made to ensure Rails 7.1.2 compatibility
- **Update `User` model serialization:** these updates produce the same default behaviour as present before the upgrade. Rails 7.1.2 now requires a coder attribute to be specified. The updated type attribute syntax avoids deprecated behaviour in the next Rails update.
- **Refactor `contact_spec` test:** Rails 7.1.2 no longer fails silently for read only attributes. This means we can now explicitly test for errors when trying to update a read only attribute and check that the attribute remains unchanged.
- **Bring the [Searchlight::Adapters::ActionView module](https://github.com/nathanl/searchlight/blob/master/lib/searchlight/adapters/action_view.rb) into the application:** This was necessary so that we can extend ActionSupport::Concerns in the module. Without this, when updating to Rails 7.1.2, searchlight no longer functions - it produces a NoMethodError when being required in the `value_objects/contacts_search` model. 

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -70,19 +79,14 @@
...
   config.active_support.report_deprecations = false
 
-  # Use default logging formatter so that PID and timestamp are not suppressed.
-  config.log_formatter = ::Logger::Formatter.new
-
-  # Use a different logger for distributed setups.
-  # require "syslog/logger"
-  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new "app-name")
-
-  if ENV["RAILS_LOG_TO_STDOUT"].present?
```
Comment (ryanb-gds) [inline config/environments/production.rb:80]: Should we be applying this env check to the new logger configuration?

Now I see it, I'm not 100% sure I did for manuals 🤔

**Review by ryanb-gds — COMMENTED** [review]
Just want to check one thing before I approve this

File: config/environments/production.rb (hunk 1 above)
Comment (jyoung-gds) [reply config/environments/production.rb:80]: Hmm good spot 🧐 I'm not totally sure about this to be honest. My gut says we should.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class User < ApplicationRecord
   include GDS::SSO::User
 
-  serialize :permissions, Array
+  serialize :permissions, type: Array, coder: YAML
```
Comment (beccapearce) [inline app/models/user.rb:4]: I think this should be JSON since that was the default

**Review by beccapearce — APPROVED** [review]

File: config/environments/production.rb (hunk 1 above)
Comment (jyoung-gds) [reply config/environments/production.rb:80]: I've decided to include this logic to be on the safe side.

**Pull request merged**